### PR TITLE
[XM] Static Registrar CreateException fix

### PR DESCRIPTION
**Pull request by timrisi**
We ran into a couple crashes while getting error info for a different crash
This lets us catch the exception if that happens and output an error for the new crash.

### CODE REVIEW

**Comment by timrisi** [comment]
Build

**Comment by chamons** [comment]
https://forums.xamarin.com/discussion/comment/256246 is an example of a user that hit this while running into some other (currently unknown) issue).

**Comment by chamons** [comment]
The other diff is there to give us more information when we crash, so we can at least know for what assembly we are dying.

@timrisi should update the commit when it is smashed.

**Comment by chamons** [comment]
@spouliot 👍  or 👎  on description? What needed before going in.

File: tools/common/StaticRegistrar.cs [hunk 1]
```diff
@@ -663,12 +663,20 @@ protected override bool HasThisAttribute (MethodDefinition method)
 
 		protected override Exception CreateException (int code, Exception innerException, MethodDefinition method, string message, params object[] args)
 		{
-			return ErrorHelper.CreateError (App, code, innerException, method, message, args);
+			try {
+				return ErrorHelper.CreateError (App, code, innerException, method, message, args);
+			} catch (Exception e) {
+				return ErrorHelper.CreateError (0099, "Internal error \"Error creating static registrar error for: {0} - {1}\" Please file a bug report with a test case (http://bugzilla.xamarin.com).", code, e);
+			}
```
Comment (spouliot) [inline tools/common/StaticRegistrar.cs]: with the above change, to `ErrorHelper.cs`, this should not be needed anymore.

File: tools/common/StaticRegistrar.cs [hunk 2]
```diff
@@ -663,12 +663,20 @@ protected override bool HasThisAttribute (MethodDefinition method)
...
 		}
 
 		protected override Exception CreateException (int code, Exception innerException, TypeReference type, string message, params object [] args)
 		{
-			return ErrorHelper.CreateError (App, code, innerException, type, message, args);
+			try {
+				return ErrorHelper.CreateError (App, code, innerException, type, message, args);
+			} catch (Exception e) {
+				return ErrorHelper.CreateError (0099, "Internal error \"Error creating static registrar error for: {0} - {1}\" Please file a bug report with a test case (http://bugzilla.xamarin.com).", code, e);
+			}
```
Comment (spouliot) [inline tools/common/StaticRegistrar.cs]: with the above change, to `ErrorHelper.cs`, this should not be needed anymore.

File: tools/common/StaticRegistrar.cs [hunk 3]
```diff
@@ -3704,8 +3712,10 @@ public void Generate (IEnumerable<AssemblyDefinition> assemblies, string header_
 		{
 			this.input_assemblies = assemblies;
 
-			foreach (var assembly in assemblies)
+			foreach (var assembly in assemblies) {
+				Driver.Log (3, "Generating static registrar for {0}", assembly.Name);
```
Comment (spouliot) [inline tools/common/StaticRegistrar.cs]: This is an unrelated change (wrt commit message). However you need to update it since it does not match the diff anymore so please also state the addition of the logging.

Comment (timrisi) [inline tools/common/StaticRegistrar.cs]: I would just update the commit message when it comes time to merge right?

Comment (spouliot) [inline tools/common/StaticRegistrar.cs]: yes :)

**Comment by timrisi** [comment]
Build
